master/internal/resourcemanagers/resource_manager_config.go

```diff
@@ -53,6 +53,17 @@ func ResolveConfig(
 		return nil, nil, errors.New(
 			"cannot specify both the scheduler and resource_manager fields")
 	}
+
+	if resourceMangerConf != nil && resourceMangerConf.AgentRM != nil && resourcePoolsConf != nil {
```

Comment:
This is handling the case where the user has not set a default CPU/GPU resource pool right? I think this could be confusing to users. I'm thinking we should make `defaultCPUResourcePool` and `defaultGPUResourcePool` required fields.

Reply:
My thought is that we should let users configure as less fields as possible. In the case that there is only resource pool, it sounds redundant to let users specify it. we haven't figured out this detail yet IIRC. we can discuss it in the sync.

Reply:
Talked offline - those fields should be required